src/client/client.ts

```diff
@@ -110,38 +110,35 @@ export class SubscriptionClient {
     this.client.close();
   }
 
-  public subscribe(options: SubscriptionOptions, handler: (error: Error[], result?: any) => void) {
-    const { query, variables, operationName, context } = options;
-
-    if (!query) {
-      throw new Error('Must provide `query` to subscribe.');
-    }
+  public query(options: RequestOptions): Promise<ExecutionResult> {
+    return new Promise((resolve, reject) => {
+      const handler = (error: Error[], result?: any) => {
+        if (result) {
+          resolve({
+            data: result,
+            errors: error,
+          });
+        } else {
+          reject(error);
+        }
+      };
 
-    if (!handler) {
-      throw new Error('Must provide `handler` to subscribe.');
-    }
+      this.executeRequest(options, handler, false);
+    });
+  }
 
-    if (
-      !isString(query) ||
-      ( operationName && !isString(operationName)) ||
-      ( variables && !isObject(variables))
-    ) {
-      throw new Error('Incorrect option types to subscribe. `subscription` must be a string,' +
-      '`operationName` must be a string, and `variables` must be an object.');
-    }
+  public subscribe(options: RequestOptions, handler: (error: Error[], result?: any) => void) {
+    const reqId = this.executeRequest(options, handler, true);
+    this.waitingSubscriptions[reqId] = true;
 
-    const subId = this.generateSubscriptionId();
-    let message = Object.assign(options, {type: SUBSCRIPTION_START, id: subId});
-    this.sendMessage(message);
-    this.subscriptions[subId] = {options, handler};
-    this.waitingSubscriptions[subId] = true;
     setTimeout( () => {
```

Comment:
the connection itself should have a timeout, if keepAlive is not heard within the timeout frame, then you should timeout, not per request..